test/e2e/upgrade/service/service.go

```diff
@@ -96,9 +96,17 @@ func (t *UpgradeTest) Setup(f *framework.Framework) {
 	// Hit it once before considering ourselves ready
 	ginkgo.By("hitting pods through the service's LoadBalancer")
 	timeout := service.LoadBalancerLagTimeoutAWS
-	// require five passing requests to continue (in case the SLB becomes available and then degrades)
-	for i := 0; i < 5; i++ {
-		service.TestReachableHTTP(tcpIngressIP, svcPort, timeout)
+	// require thirty seconds of passing requests to continue (in case the SLB becomes available and then degrades)
+	for successCount := 0; successCount < 30; successCount++ {
+		// sleep for 1s between successive tries.
+		time.Sleep(1 * time.Second)
+		func() {
+			defer func() {
+				recover()
+				successCount = 0
+			}()
+			service.TestReachableHTTP(tcpIngressIP, svcPort, timeout)
```

Comment:
Unfortunately not the same, the loop I was talking about hoisting was TestReachableHTTPWithRetriableErrorCodes

```
func TestReachableHTTPWithRetriableErrorCodes(host string, port int, retriableErrCodes []int, timeout time.Duration) {
	pollfn := func() (bool, error) {
		result := e2enetwork.PokeHTTP(host, port, "/echo?msg=hello",
			&e2enetwork.HTTPPokeParams{
				BodyContains:   "hello",
				RetriableCodes: retriableErrCodes,
			})
		if result.Status == e2enetwork.HTTPSuccess {
			return true, nil
		}
		return false, nil // caller can retry
	}

	if err := wait.PollImmediate(framework.Poll, timeout, pollfn); err != nil {
		if err == wait.ErrWaitTimeout {
			framework.Failf("Could not reach HTTP service through %v:%v after %v", host, port, timeout)
		} else {
			framework.Failf("Failed to reach HTTP service through %v:%v: %v", host, port, err)
		}
	}
}
```

Reply:
just take all of this code and move it up into a function in this, and call it TestReachableWithMinimumSuccess

Reply:
I didn't want to duplicate code especially since this is vendored from k/k. Do you think the above loop will work?

Reply:
I want to duplicate code.  Kube code isn't special.  We already split half the e2es because the upstream code is unreliable and not fit for purpose.

Reply:
Ok, I will create a separate function like `TestReachableWithMinimumSuccess` and use that.